lib/validate.js

```diff
@@ -124,7 +124,34 @@ const validateCategory = (
     const isSchemaProp = schemaProps.has(prop);
     const isObjectProp = objectProps.has(prop);
     if (isObjectProp && !isSchemaProp) {
-      errors.push(new ValidationError('unresolvedProperty', `${path}${prop}`));
+      if (schema[Symbol.for('other')]) {
+        const propPath = `${path}${prop}`;
+        const valid = schema[Symbol.for('other')].some(type => {
+          let kind, name;
```

Comment:
IMHO you need to rename the type and kind to something more meaningful. Means the same thing.